### PR TITLE
Add migration to update enrollment code product class.

### PR DESCRIPTION
**Pull request by vkaracic**
https://openedx.atlassian.net/browse/ECOM-7393

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @vkaracic! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fecommerce&number=1191"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

**Review by clintonb — COMMENTED** [review]
This is unnecessary. There is nothing to be consistent with given that we don't use slugs. Also, the value in production is `enrollment_code`.

**Comment by vkaracic** [comment]
Yes, but the value in every sandbox from here on out is going to be ``enrollment-code`` which might cause confusion if someone decides to use the slug attribute.

Not to mention that it's probably going to be like that in any possible open source contribution.

**Comment by vkaracic** [comment]
AND in every new fresh installation of ecommerce.

File: ecommerce/extensions/catalogue/migrations/0024_fix_enrollment_code_slug.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+Product = get_model("catalogue", "Product")
+ProductClass = get_model("catalogue", "ProductClass")
+
+WRONG_SLUG = 'enrollment-code'
+RIGHT_SLUG = 'enrollment_code'
+
+
+def fix_enrollment_code_slug(apps, schema_editor):
+    """Update or remove the faulty product class and update any enrollment codes with faulty product class."""
+    faulty_product_class = ProductClass.objects.filter(slug=WRONG_SLUG).first()
```
Comment (clintonb) [inline ecommerce/extensions/catalogue/migrations/0024_fix_enrollment_code_slug.py]: Keep it simple. We don't want to touch every product, or delete anything.

```python
try:
  product_class = ProductClass.objects.get(slug=WRONG_SLUG)
  product_class.slug = RIGHT_SLUG
  product_class.save()
except DoesNotExist:
  # If it's not broken, there is nothing to fix.
  pass
```

The revert can be the same, but swap the constant names.

**Review by clintonb — APPROVED** [review]

**Review by marjev — APPROVED** [review]

**Pull request merged**